plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/mapping/CacheBasedIdentifierMapping.java

```diff
@@ -143,28 +148,67 @@ public String toRemoteColumnName(Connection connection, String columnName)
         return defaultIdentifierMapping.toRemoteColumnName(connection, columnName);
     }
 
-    private Map<String, String> createSchemaMapping(Connection connection)
+    private Mapping createSchemaMapping(Connection connection)
     {
-        return baseJdbcClient.get().listSchemas(connection).stream()
-                // will throw on collisions to avoid ambiguity
-                .collect(toImmutableMap(
-                        remoteSchemaName -> defaultIdentifierMapping.fromRemoteSchemaName(remoteSchemaName),
-                        schemaName -> schemaName));
+        Map<String, String> map = new HashMap<>();
```

Comment:
I think most of this code (apart from retrieving the list of schemas) can be moved to the constructor of `Mapping`. Where the data structures could be documented and explained :)

Reply:
Especially given that the logic is duplicated for tables as well.

Reply:
I don't wan't to put much logic to constructor. Will extract a method.